redbot/cogs/streams/streamtypes.py

```diff
@@ -280,92 +282,101 @@ def __init__(self, **kwargs):
-                    "https://api.twitch.tv/helix/users/follows", headers=header, params=params
-                ) as r:
-                    user_data = await r.json(encoding="utf-8")
-            if user_data:
-                followers = user_data["total"]
-                data["followers"] = followers
-
-            params = {"id": self.id}
-            async with aiohttp.ClientSession() as session:
+            try:
                 async with session.get(
-                    "https://api.twitch.tv/helix/users", headers=header, params=params
-                ) as r:
-                    user_profile_data = await r.json(encoding="utf-8")
-            if user_profile_data:
-                profile_image_url = user_profile_data["data"][0]["profile_image_url"]
-                data["profile_image_url"] = profile_image_url
-                data["view_count"] = user_profile_data["data"][0]["view_count"]
-                data["login"] = user_profile_data["data"][0]["login"]
-
-            is_rerun = False
-            return self.make_embed(data), is_rerun
-        elif r.status == 400:
+                    url, headers=header, params=params, timeout=aiohttp.ClientTimeout(total=None)
+                ) as resp:
+                    remaining = resp.headers.get("Ratelimit-Remaining")
+                    if remaining:
+                        self._rate_limit_remaining = int(remaining)
+                    reset = resp.headers.get("Ratelimit-Reset")
+                    if reset:
+                        self._rate_limit_resets.add(int(reset))
+
+                    if resp.status == 429:
+                        log.info(
+                            "Ratelimited. Trying again at %s.", datetime.fromtimestamp(int(reset))
+                        )
+                        return await self.get_data(url)
+
+                    if resp.status != 200:
+                        return None, {}
+
+                    return resp.status, await resp.json(encoding="utf-8")
+            except aiohttp.ClientConnectionError:
+                return None, {}
+
+    async def is_online(self):
+        user_profile_data = await self._fetch_user_profile()
+        final_data = {
+            "game_name": None,
+            "followers": None,
+            "view_count": None,
+            "profile_image_url": None,
+            "login": None,
+            **user_profile_data,
+        }
+
+        stream_code, stream_data = await self.get_data(
+            TWITCH_STREAMS_ENDPOINT, {"user_id": self.id}
+        )
+        if stream_code == 200:
+            if not stream_data["data"]:
+                raise OfflineStream()
+            stream_data = stream_data["data"][0]
+            self.name = stream_data["user_name"]
+            final_data = {**final_data, **stream_data}
+
+            follows_data = await self.get_data(TWITCH_FOLLOWS_ENDPOINT, {"to_id": self.id})
+            final_data["followers"] = follows_data[1]["total"]
```

Comment:
Before, this code ensured that follows data is not empty and it should continue to do that.
Also, let's use tuple unpacking to simplify this a bit.
```suggestion
            __, follows_data = await self.get_data(TWITCH_FOLLOWS_ENDPOINT, {"to_id": self.id})
            if follows_data:
                final_data["followers"] = follows_data["total"]
```